### ISSUE TITLE
False positive error message: Match must be exhaustive [E0004]

### ISSUE THREAD

**Issue by StPanning** [intellij-rust/intellij-rust#9373]
## Environment

* **IntelliJ Rust plugin version:** 0.4.178.4873-222
* **Rust toolchain version:** 1.63.0 (4b91a6ea7 2022-08-08) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** Linux 5.15.0-47-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Following code compiles, but IDE produces error-message "Match must be exhaustive [E0004]"

## Steps to reproduce
```rust
enum SubOp {
   X(i32),
   Y(i32),
}

enum Optype1 {
   A(SubOp),
   B(SubOp),
}

fn main() {
   let join = Optype1::A(SubOp::X(1));
   match join {
      Optype1::A(SubOp::X(qt))
      | Optype1::A(SubOp::Y(qt))
      | Optype1::B((SubOp::X(qt)))
      | Optype1::B((SubOp::Y(qt))) => {
         ()
      }
   }
}
```
